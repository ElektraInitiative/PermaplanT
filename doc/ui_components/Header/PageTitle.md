# PageTitle

## Description

Displays an <h1> heading.

## Props

| Name  | Type   | Optional | Description                                    |
| ----- | ------ | -------- | ---------------------------------------------- |
| title | string | no       | The text that will be displayed in the heading |
