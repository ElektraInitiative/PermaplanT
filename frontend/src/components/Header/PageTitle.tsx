interface PageTitleProps {
  /** Page title text content. */
  title: string;
}

/** A simple custom-styled page header. */
const PageTitle = ({ title }: PageTitleProps) => {
  return <h1 className="mb-8 text-2xl">{title}</h1>;
};

export default PageTitle;
