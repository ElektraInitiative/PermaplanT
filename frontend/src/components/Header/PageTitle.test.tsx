import PageTitle from './PageTitle';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <PageTitle title={'Hello World!'} />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
