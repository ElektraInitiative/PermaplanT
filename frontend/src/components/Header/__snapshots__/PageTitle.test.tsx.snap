// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<h1
  className="mb-8 text-2xl"
>
  Hello World!
</h1>
`;
